eurocodedesign/stepper/__init__.py

```diff
@@ -0,0 +1,71 @@
+"""
+Thread-safe module for collecting printable information on calculation steps
+
+Examples
+    from eurocodedesign import stepper
+    >>> with stepper.create() as st1:
+    >>>    section = "IPE300"
+    >>>    st1.step(f"Steel section {section} has height 3.0 m.")
+    >>>    # do stuff
+    >>>    st1.step(f"{section} is of cross-section type {QK}.")
+    Steel section IPE300 has height 3.0 m. IPE300 is of cross-section type QK3.
+    >>> st2 = stepper.create()
+    >>> st2.step("Add step")
+    >>> del st2
+    Add step
+"""
+from collections import deque
+from typing import Self
+
+
+class Stepper:
+    """
+    Thread-safe context manager for collection calculation steps
+    """
+    _steps: deque
+
+    def __init__(self) -> None:
+        self._steps = deque()
+
+    def __str__(self) -> str:
+        """
+        Returns: Added steps concatenated to one string separated by spaces.
+        """
+        if self._steps:
+            return ' '.join(self._steps)
+        return ''
+
+    def step(self, description: str) -> None:
```

Comment:
I think it would be useful to have `step` return the current state of the string, or probably better a `get` method that returns the current state of `Stepper._steps`.

I think this would provide more flexibility in the future. Am I right in thinking that the only way of seeing what is stored in `Stepper._steps` is to call `print(Stepper)`?

Reply:
I'm thinking about long notebooks of calculations where we may want to print relevant info to screen... At each point we would use a new stepper function for the just completed calculation. However, it would be nicre to be able to save the text and then reuse it at the end of the notebook to create a final .pdf report. Being able to save the assign the value of `Stepper._steps` would be useful.

So would being able to create a new `Stepper` instance with a list of strings --- idea for future issue perhaps

Reply:
Yes, I would do this in a later issue. Maybe with a constructor which allows multiple other stepper instances as parameters.